src/style/style_layer.js

```diff
@@ -32,6 +38,7 @@ class StyleLayer extends Evented {
     maxzoom: ?number;
     filter: FilterSpecification | void;
     visibility: 'visible' | 'none';
+    transitionParameters: EvaluationParameters;
```

Comment:
Prefix with `_` to indicate that this is private.  Maybe also rename to just `_parameters`, since this is used to evaluate all zoom-driven properties, not just transitionable ones.